src/dag.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-env mocha */
+/* eslint max-nested-callbacks: ["error", 8] */
+
+'use strict'
+
+const expect = require('chai').expect
+const dagPB = require('ipld-dag-pb')
+const DAGNode = dagPB.DAGNode
+const dagCBOR = require('ipld-dag-cbor')
+
+module.exports = (common) => {
+  describe.only('.dag (basics)', () => {
+    let ipfs
+
+    before(function (done) {
+      // CI is slow
+      this.timeout(20 * 1000)
+
+      common.setup((err, factory) => {
+        expect(err).to.not.exist
+        factory.spawnNode((err, node) => {
+          expect(err).to.not.exist
+          ipfs = node
+          done()
+        })
+      })
+    })
+
+    after((done) => {
+      common.teardown(done)
+    })
+
+    describe('callback API', () => {
+      let pbNode
+      let cborNode
+
+      before((done) => {
+        const someData = new Buffer('some data')
+
+        pbNode = DAGNode.create(someData, (err, node) => {
+          expect(err).to.not.exist
+          pbNode = node
+          done()
+        })
+
+        cborNode = {
+          data: someData
+        }
+      })
+
+      describe('.put', () => {
```

Comment:
that is what happens in the get tests, this way put is tested in isolation and in conjunction with get